credential/src/main/java/io/syndesis/credential/Credentials.java

```diff
@@ -45,56 +38,43 @@
 import org.springframework.util.Assert;
 import org.springframework.util.LinkedMultiValueMap;
 import org.springframework.util.MultiValueMap;
-import org.springframework.web.context.request.NativeWebRequest;
-import org.springframework.web.context.request.RequestAttributes;
 import org.springframework.web.util.UriComponentsBuilder;
 
 @Component
 public final class Credentials {
-
-    public static final String CACHE_NAME = "credentials";
-
-    private static final Pattern API_BASE_PATH = Pattern.compile("(/[^/]+/[^/]+)/.*");
-
     private static final MultiValueMap<String, String> EMPTY = new LinkedMultiValueMap<>();
 
-    protected static final String CREDENTIAL_FLOW_STATE_SESSION_KEY = "credentialFlowState";
-
     private final CredentialProviderLocator credentialProviderLocator;
 
     private final DataManager dataManager;
 
-    private final Cache state;
-
     @Autowired
-    public Credentials(final CredentialProviderLocator credentialProviderLocator, final DataManager dataManager,
-        final CacheManager cacheManager) {
+    public Credentials(final CredentialProviderLocator credentialProviderLocator, final DataManager dataManager) {
         this.credentialProviderLocator = credentialProviderLocator;
         this.dataManager = dataManager;
-        state = cacheManager.getCache(CACHE_NAME);
     }
 
-    public Acquisition acquire(final String connectionId, final String providerId, final URI returnUrl,
-        final NativeWebRequest request) {
+    public AcquisitionFlow acquire(final String connectionId, final String providerId, final URI baseUrl,
+        final URI returnUrl) {
 
         final ConnectionFactory<?> connectionFactory = factoryFor(providerId);
 
         final String redirectUrl;
-        final CredentialFlowState.Builder flowStateBuilder = new CredentialFlowState.Builder().returnUrl(returnUrl)
-            .connectionId(connectionId).providerId(providerId);
+        final CredentialFlowState.Builder flowStateBuilder = CredentialFlowState.builderFor(connectionFactory)
+            .returnUrl(returnUrl).connectionId(connectionId).providerId(providerId);
 
         if (connectionFactory instanceof OAuth1ConnectionFactory) {
-            redirectUrl = prepareOAuth1((OAuth1ConnectionFactory<?>) connectionFactory, flowStateBuilder, request);
+            redirectUrl = prepareOAuth1((OAuth1ConnectionFactory<?>) connectionFactory,
```

Comment:
Why not move all these `prepareOAuth{1,2}` and similar methods into relevant connection factories?